Support for multi-cursor alignment

csholmq opened the pull request:
- Added cursorAlign
- Renamed block to blockAlign
- Updated README
- Bumped to 0.1.0

Addresses #1

csholmq committed:
Added cursorAlign. Renamed block to blockAlign.

csholmq committed:
Bumped version to 0.1.0

csholmq committed:
Updated README

blakeembrey commented:
This will need some tests. Is there a reason for the API changes? Instead of expecting text, it now gets lines. Instead of cursor positions as `[line, col]` they're a `position`?

blakeembrey commented:
What about handling selections? It'd be nice to just offload text and selections, and get back new text and new cursor positions.

You're welcome to disagree though, from the PR I understand it expects cursors to be in very static positions. What about two cursors on a line?

csholmq commented:
Thanks for the feedback. I'll take a look tomorrow night.

csholmq commented:
Based on your feedback, I reverted the API changes (i.e. renamed `blockAlign` back to `block`).

The idea behind having `lines: string[]` and relative `cursors: number[]` was that a user might select rows far apart from one another. So instead of processing a large chunk of text, slice, align, join, etc. all the function would do is calculate the required padding length for each line and then it's up to the application on how to pad and replace the selections. Thus, the return of the text is not neccessary.

Multiple cursors on one line doesn't make sense and I fail to see how that could be supported.

blakeembrey commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "alignment",
-  "version": "0.0.1",
+  "version": "0.1.0",
```
Please don't bump versions on people, I would have to rebase it to remove this.

csholmq replied:
Roger. Bad habit as multiple extension owners have asked me to bump version.

blakeembrey commented on README.md:
````diff
@@ -32,6 +31,17 @@ console.log(result)
+```js
+import { cursorAlign } from 'alignment'
+
+const padLen: CursorResult = cursorAlign(lines, cursors)
````
The previous examples were just JavaScript and not TypeScript specific.

blakeembrey commented:
@csholmq Thanks for your work, I'm going to look at adding it to a release and make it handle multiple cursors per line, etc better - sorry for the delay!

csholmq commented:
@blakeembrey Ok, no worries. I'll close this then. Reopen if required.

blakeembrey commented:
@csholmq So I did spend a day (cumulative, I had a lot of distractions) and tried to fix this to handle multiple columns. It's definitely possible, and I can outline my approach which you're free to attempt it. I realise it's a chunk of work and I may not complete it in the next few weeks. I'd prefer to provide a complete solution though.

So, which I did was create a `cursor (text: string, cursors: Cursor[]): AlignmentResult` function. It took the cursors and turned them into a matrix of columns. From that I was normalizing the columns to the max positions in each that would be required to align. After that, we need to go over the text and add whitespace equal to the old position and new position diff. Finally return the normalizing columns and aligned text. What do you think?
